### DISCUSSION TITLE
zoraxy using Amazon Route 53

### DISCUSSION THREAD

**Discussion by achimwessling** [tobychui/zoraxy#369] [Q&A]
Hey,
I setup zoraxy as a docker container with docker compose. It seams to work fine. I can login and do basic configuration. But I get an error when I try to configure the DNS challenge for my Route53 domain.
![Bildschirmfoto vom 2024-11-02 09-56-50](https://github.com/user-attachments/assets/be24bf93-fb61-420b-839b-eb7a67b02a48)

First I get the information "Certifikate DNS Credentials updated successfully", but it's followed by "Json: cannot unmarshal string into Go struct field Config.MaxRetries of type int". Can anyone help?
![Bildschirmfoto vom 2024-11-02 10-08-32](https://github.com/user-attachments/assets/051c8c3b-1d29-4a17-833e-4783dc1462e4)
![Bildschirmfoto vom 2024-11-02 10-08-19](https://github.com/user-attachments/assets/dda5aad8-af53-4e2a-b4db-7f7cf3966e70)

**Comment by tobychui** [comment]
@achimwessling I didn't use Route 53 so I am not sure about that, but I guess you can try removing the following field in the `mod/acme/acmedns/providers.json` and use `go build` to build your own Zoraxy to see if this is fixed?

Remove this in `providers.json`
```json
{
    "Title": "MaxRetries",
    "Datatype": "int"
   },
```

It also might be a front-end problem. You can try modify the form render and submission function as follows and see if this is fixed.
In `acme.html` Line 438, add typeint to `$("#dnsProviderAPIFields").append(`<div class="ui fluid labeled dnsConfigField...` line's class name
```js
for (const [key, datatype] of Object.entries(data)) {
          if (datatype == "int"){
            let defaultValue = 10;
            if (key == "HTTPTimeout"){
              defaultValue = 300;
            }
            $("#dnsProviderAPIFields").append(`<div class="ui fluid labeled dnsConfigField input typeint" key="${key}" style="margin-top: 0.2em;">
              <div class="ui basic blue label" style="font-weight: 300;">
                ${key}
              </div>
              <input type="number" value="${defaultValue}">
            </div>`);
          }else if (datatype == "bool"){
```

and replace the `readDnsCredentials()` function with this new parser
```js
function readDnsCredentials(){
      let dnsCredentials = {};
      $(".dnsConfigField").each(function(){
        let thisKey = $(this).attr("key");
        let value = "";
        if ($(this).hasClass("checkbox")){
          //Boolean option
          let checked = $(this).find("input")[0].checked;
          dnsCredentials[thisKey] = checked;
        }else if ($(this).hasClass("typeint")){
          //Int options
          let value = $(this).find("input").val();
          dnsCredentials[thisKey] = parseInt(value);
        }else{
          //String options
          let value = $(this).find("input").val().trim();
          dnsCredentials[thisKey] = value;
        }
      });
      
      return dnsCredentials;
    }
```
